Automattic/wp-calypso
# A4A > Referrals: Implement client auth

Closes https://github.com/Automattic/jetpack-genesis/issues/374

## Proposed Changes

This PR 

- Implements the client authentication UI for A4A Referrals.
- Removes some code that are no more required.

NOTE: 

- [Design](https://www.figma.com/design/fuufP6VNfZXYmvLsTRWRlG/Referrals?node-id=6711-71510&m=dev)

This is the client auth implementation. Currently, the agency users can see the client UI for testing purposes. We will disable it once everything is implemented. While this is not the ideal way to implement this. We decided to go ahead with this for the MVP (CC @vitozev)

## Testing Instructions

1. Check out the branch and start the server.
2. Set the `isClient` to false here: https://github.com/Automattic/wp-calypso/pull/91640/files#diff-90a95b186e56a1d11455fff699d8237d3fa7bb6da1d8eba9583518ead6be10b5R32
3. Visit `/client/subscriptions` & verify you are redirected to /overview.
4. Now, set the `isClient` to true > Visit `/client/subscriptions` 
5. Verify that you see the page below, which includes two menu items: `Your Subscriptions` and `Payment Methods`.Ideally, you will be directed to /overview if you have an agency account.


<img width="433" alt="Screenshot 2024-05-31 at 10 13 02 AM" src="https://github.com/Automattic/wp-calypso/assets/10586875/515d5d36-739e-407c-95b3-6f114d7ad85a">

## Pre-merge Checklist
